static/js/workers/vfs.worker.js

```diff
@@ -0,0 +1,711 @@
+   *
+   * @param {string} path - The file path.
+   * @param {object} content - The new content of the file.
+   * @param {boolean} [isUserInvoked] - Whether to user invoked the action.
+   * @returns {FileSystemFileHandle} The updated file handle.
+   */
+  async writeFile({ path, content, isUserInvoked }) {
+    console.log(`writeFile: ${path} ${isUserInvoked}`);
+    const handle = await getFileHandleByPath(path);
+    if (!handle) return;
+
+    // TODO where should debouncing go? See original.
+
+    if (isOPFS()) {
+      // use Safari-compatible API
+      console.log("writeFile to OPFS");
+      const accessHandle = await handle.createSyncAccessHandle();
+      const data = new TextEncoder().encode(content);
+      accessHandle.truncate(data.byteLength);
+      accessHandle.write(data, { at: 0 });
+      accessHandle.flush();
+      accessHandle.close();
+    } else {
+      // use general FS API
+      const writable = await handle.createWritable();
+      await writable.write(content);
+      await writable.close();
+    }
+
+    if (isUserInvoked) {
+      console.log("user invoked change");
+      self.postMessage({
+        type: "fileContentChanged",
+        data: { detail: { file: { path, content } } },
+      });
+    }
+  },
+
+  /**
+   * Delete a file.
+   *
+   * @param {string} id - The path of the file to delete.
+   * @param {boolean} [isSingleFileDelete] - whether this function is called for
+   * a single file or is called from the `deleteFolder` function.
+   * @returns {boolean} True if deleted successfully, false otherwise.
+   */
+  async deleteFile({ path, isUserInvoked = true }) {
+    if (!(await pathExists(path))) {
+      return false;
+    }
+
+    const parts = path.split("/");
+    const filename = parts.pop();
+    const parentPath = parts.join("/");
+    const parent = await getFolderHandleByPath(parentPath);
+    await parent.removeEntry(filename);
+
+    // if (isUserInvoked) {
+    //   this.dispatchEvent(
+    //     new CustomEvent("fileDeleted", {
+    //       detail: {
+    //         file: { path },
+    //       },
+    //     }),
+    //   );
+    // }
+
+    // TODO
+    return { ok: true };
+  },
+
+  /**
+   * Gathers all files from the VFS.
+   * Formerly known as getAllEditorFiles.
+   *
+   * @async
+   * @param {string} [folderpath=''] - The folder path to start searching from.
+   * @returns {Promise<object[]>} List of objects, each containing the filepath
+   * and content of the corresponding file.
+   */
+  async getAllFiles() {
+    console.log(`getAllFiles:`);
+    const root = await getRootHandle();
+    const files = [];
+
+    async function walk(folderHandle, currentPath = "") {
+      for await (const [name, handle] of folderHandle.entries()) {
+        if (blacklistedPaths.includes(name)) continue;
+        const path = currentPath ? `${currentPath}/${name}` : name;
+
+        if (handle.kind === "file") {
+          const accessHandle = await handle.createSyncAccessHandle();
```

Comment:
Op safari kan je gewoon op een FileSystemFileHandle direct `file = handle.getFile()` aanroepen en daarop weer `file.text()` om de content te krijgen, dus als het goed is zó veel code niet nodig.